### ISSUE TITLE
Compact mode: back button click animation leaves persistent background color artifacts

### ISSUE THREAD

**Issue by foresto** [Nheko-Reborn/nheko#1153]
### Describe the bug

In compact mode, the back button in a room's upper left corner is often drawn with the wrong background color, and sometimes with a non-square background shape.

### To Reproduce

1. Make the Nheko window small enough for compact mode to engage.
2. Enter a room.
3. Exit the room.
4. Re-enter the room.
5. Look at the back button in the upper left corner.

### What happened?

The button's background does not match the surrounding area or the desktop color theme. Instead, it is usually a gray square, and occasionally a different shape that looks like the edge of a gray disc within a square clipping area.

Clicking the button and then re-entering the room sometimes changes the button's background color or shape. The problem doesn't appear for me until the second time I enter the room. Try exiting & re-entering several times if you don't see it on the second try.

### Expected behavior

The button's background color should match the surrounding area, as it does in the settings screen.

### Screenshots

![nheko-compact-edit-grey-box](https://user-images.githubusercontent.com/322134/184565452-4e30a86c-92ab-4a89-869f-ff19bb829a05.jpg)

### Version

0.10.0

### Operating system

Linux

### Installation method

Locally built flatpak using the Flathub .yaml file

### Qt version

5.15.2

### C++ compiler

whatever flatpak-builder 1.0.12 uses on debian stable (bullseye)

### Desktop Environment

KDE Plasma 5.20.5

### Did you use profiles?

- [ ] Profiles used?

**Comment by foresto** [comment]
I just noticed that the strange background color seems to be a click animation effect:

When I click the back button, a small grey disc appears centered on the click point. It then expands outward from that point, clipped to the area of the button's box.

Apparently the disc never gets removed after the animation completes, so when I return to a room, the disc is still visible in its expanded state. If I clicked near a corner of the back button, the disc's curve is visible near the opposite corner. If I clicked in the center of the button, the disc reaches beyond the bounding box in all directions, so it looks like a square.
